src/ncp/ncp_base.cpp

```diff
@@ -6683,35 +6680,30 @@ otError NcpBase::InsertPropertyHandler_THREAD_OFF_MESH_ROUTES(uint8_t aHeader, s
                        &flags
                    );
 
-    if (parsedLength > 0)
-    {
-        routeConfig.mPrefix.mPrefix = *addrPtr;
-        routeConfig.mStable = stable;
-        routeConfig.mPreference = FlagByteToExternalRoutePreference(flags);
+    VerifyOrExit(parsedLength > 0, spinelError = SPINEL_STATUS_PARSE_ERROR);
 
-        error = otBorderRouterAddRoute(mInstance, &routeConfig);
+    routeConfig.mPrefix.mPrefix = *addrPtr;
+    routeConfig.mStable = stable;
+    routeConfig.mPreference = FlagByteToExternalRoutePreference(flags);
 
-        if (error == OT_ERROR_NONE)
-        {
-            error = SendPropertyUpdate(
-                        aHeader,
-                        SPINEL_CMD_PROP_VALUE_INSERTED,
-                        aKey,
-                        aValuePtr,
-                        aValueLen
-                    );
-        }
-        else
-        {
-            error = SendLastStatus(aHeader, ThreadErrorToSpinelStatus(error));
-        }
-    }
-    else
+    error = otBorderRouterAddRoute(mInstance, &routeConfig);
+    VerifyOrExit(error == OT_ERROR_NONE, spinelError = ThreadErrorToSpinelStatus(error));
+
+    error = SendPropertyUpdate(
```

Comment:
This PR will now call `SendLastStatus()` if `SendPropertyUpdate()` returns an error.  I assume this is an intentional change.